syntax = "proto3";

package network;

message Heartbeat {
    repeated PlayerPosition players = 1;
    repeated MonsterPosition monsters = 2;
}

message WorldData {
    repeated PlayerInitialData players = 1;
    repeated MonsterInitialData monsters = 2;
}

message RequestLogin {
    uint32 player_id = 1;
}

message Response {
    string data = 1;
}

message ClientLogout{
    uint32 player_id = 1;
}

message ClientLogin{
    PlayerInitialData player_data = 1;
}

message Wrapper {
    enum MessageType {
        WORLDDATA = 0;
        REQUESTLOGIN = 1;
        CLIENTLOGOUT = 2;
        CLIENTLOGIN = 3;
        PLAYERINITIALDATA = 4;
        PLAYERPOSITION = 5;
        HEARTBEAT = 6;
        RESPONSE = 100;
    }
    MessageType type = 1;
    bytes payload = 2;
}

message AssignId {
    uint32 id = 1;
}

// Characters Initial Data

message PlayerInitialData {
    uint32 player_id = 1;
    string player_name = 2;
    float position_x = 3;
    float position_y = 4;
    float position_z = 5;
    uint32 rotation_y = 6;
    uint32 player_movementspeed = 7;
}

// Characters Position

message PlayerPosition {
    uint32 player_id = 1;
    float position_x = 2;
    float position_y = 3;
    float position_z = 4;
    uint32 rotation_y = 5;
}
//Monsters Initial Data

message MonsterInitialData {
        int32 monster_id = 1;
    string monster_name = 2;
    float position_x = 3;
    float position_y = 4;
    float position_z = 5;
    uint32 rotation_y = 6;
    uint32 monster_movement_speed = 7;
}

// Monsters Position
message MonsterPosition {
    int32 monster_id = 1;
    string monster_name = 2;
    float position_x = 3;
    float position_y = 4;
    float position_z = 5;
    uint32 rotation_y = 6;
}

//NetAnimations
message AnimationClip{

}